components/inputs/input-number.js

```diff
@@ -331,6 +337,7 @@ class InputNumber extends FocusMixin(LabelledMixin(SkeletonMixin(FormElementMixi
 	}
 
 	render() {
+		const valueAlign = (this.valueAlign === 'end') ? 'end' : 'start';
```

Comment:
The validation stuff hates it if you just assign an arbitrary value to an enum field. This seems to help it.